kube-runtime/src/controller/mod.rs

```diff
@@ -758,6 +758,62 @@ where
         self
     }
 
+    /// Trigger the reconciliation process for a managed object `ObjectRef<K>` whenever `trigger` emits a value
+    ///
+    /// For example, this can be used to watch resources once and use the stream to trigger reconciliation and also keep a cache of those objects.
+    /// That way it's possible to use this up to date cache instead of querying Kubernetes to access those resources
+    ///
+    /// Example:
+    ///
+    /// ```no_run
+    /// # async {
+    /// use futures::{StreamExt, TryStreamExt};
+    /// use k8s_openapi::api::core::v1::{ConfigMap, Pod};
+    /// # use kube::api::{Api, ListParams};
+    /// # use kube::runtime::controller::Action;
+    /// # use kube::runtime::reflector::ObjectRef;
+    /// # use kube::runtime::{reflector, watcher, Controller, WatchStreamExt};
+    /// # use kube::{Client, ResourceExt};
+    /// use std::convert::Infallible;
+    /// use std::future;
+    /// use std::sync::Arc;
+    ///
+    /// # let client: Client = todo!();
```

Comment:
you can hide some functions here somewhere to make the doc tests pass i think

```suggestion
    /// # let client: Client = todo!();
    /// async fn reconcile(_: Arc<ConfigMap>, _: Arc<Store<Pod>>) -> Result<Action, Error> { Ok(Action::await_change()) }
    /// fn error_policy(_: Arc<ConfigMap>, _: &kube::Error, _: Arc<Store<Pod>>) -> Action { Action::await_change()) }
```

might have to do it outside the async wrapper tho